sys/linux/rdma.txt

```diff
@@ -0,0 +1,1464 @@
+include <uapi/rdma/ib_user_ioctl_cmds.h>
+include <uapi/rdma/mlx5_user_ioctl_cmds.h>
+
+# resources
+resource fd_rdma[fd]
+resource pd_handle[int32]
+resource ah_handle[int32]
+resource mr_handle[int32]
+resource mr_rkey[int32]
+resource mr_lkey[int32]
+resource cq_handle[int32]
+resource qp_handle[int32]
+resource qp_number[int32]
+resource mw_handle[int32]
+resource srq_handle[int32]
+resource xrcd_handle[int32]
+resource wq_handle[int32]
+resource ind_tbl_handle[int32]
+resource flow_handle[int32]
+
+# defines
+define IB_USER_VERBS_EX_CMD_QUERY_DEVICE	0x80000001
+define IB_USER_VERBS_EX_CMD_CREATE_FLOW	0x80000032
+define IB_USER_VERBS_EX_CMD_DESTROY_FLOW	0x80000033
+define MLX5_CREATE_DCT	0x8
+define MLX5_CREATE_DCI	0x16
+define IB_USER_VERBS_CMD_FLAG_EXTENDED	0x80000000
+define IB_USER_VERBS_CMD_FLAGS_SHIFT	24
+define IB_USER_VERBS_CMD_THRESHOLD	50
+define IB_USER_VERBS_CMD_CREATE_CQ	0x12
+define IB_USER_VERBS_CMD_CREATE_QP	0x18
+define EX_CREATE_CQ_CMD	IB_USER_VERBS_CMD_FLAG_EXTENDED | IB_USER_VERBS_CMD_CREATE_CQ
+define EX_CREATE_QP_CMD	IB_USER_VERBS_CMD_FLAG_EXTENDED | IB_USER_VERBS_CMD_CREATE_QP
+
+# flags
+rdma_dev_open_flags = O_RDWR, O_CLOEXEC
+ib_access_flags = IB_ACCESS_LOCAL_WRITE, IB_ACCESS_REMOTE_WRITE, IB_ACCESS_REMOTE_READ, IB_ACCESS_REMOTE_ATOMIC, IB_ACCESS_MW_BIND, IB_ZERO_BASED, IB_ACCESS_ON_DEMAND
+ib_mr_rereg_flags = IB_MR_REREG_TRANS, IB_MR_REREG_PD, IB_MR_REREG_ACCESS, IB_MR_REREG_SUPPORTED
+ib_qp_type = IB_QPT_SMI, IB_QPT_GSI, IB_QPT_RC, IB_QPT_UC, IB_QPT_UD, IB_QPT_RAW_IPV6, IB_QPT_RAW_ETHERTYPE, IB_QPT_RAW_PACKET, IB_QPT_XRC_INI, IB_QPT_XRC_TGT
+ib_qp_create_flags = IB_QP_CREATE_IPOIB_UD_LSO, IB_QP_CREATE_BLOCK_MULTICAST_LOOPBACK, IB_QP_CREATE_CROSS_CHANNEL, IB_QP_CREATE_MANAGED_SEND, IB_QP_CREATE_MANAGED_RECV, IB_QP_CREATE_NETIF_QP, IB_QP_CREATE_SIGNATURE_EN, IB_QP_CREATE_SCATTER_FCS
+ib_send_flags = IB_SEND_FENCE, IB_SEND_SIGNALED, IB_SEND_SOLICITED, IB_SEND_INLINE, IB_SEND_IP_CSUM
+ib_flow_flags = IB_FLOW_ATTR_FLAGS_DONT_TRAP
+ib_ipv4_flags = IB_IPV4_DONT_FRAG, IB_IPV4_MORE_FRAG
+mlx5_create_qp_flags = MLX5_QP_FLAG_SIGNATURE, MLX5_QP_FLAG_SCATTER_CQE, MLX5_QP_FLAG_TUNNEL_OFFLOADS, MLX5_QP_FLAG_BFREG_INDEX, MLX5_CREATE_DCT, MLX5_CREATE_DCI
+mlx5_create_cq_flags = 0
+create_cq_ex_flags = IB_UVERBS_CQ_FLAGS_TIMESTAMP_COMPLETION, IB_UVERBS_CQ_FLAGS_IGNORE_OVERRUN
+create_cq_ex_mask = 0x0, 0x1
+mlx5_comp_cqe_res_format = MLX5_IB_CQE_RES_FORMAT_HASH, MLX5_IB_CQE_RES_FORMAT_CSUM
+wq_type = IB_WQT_RQ
+create_wq_flags = IB_WQ_FLAGS_CVLAN_STRIPPING, IB_WQ_FLAGS_SCATTER_FCS, IB_WQ_FLAGS_DELAY_DROP, IB_WQ_FLAGS_PCI_WRITE_END_PADDING
+modify_wq_attr_flags = IB_WQ_STATE, IB_WQ_CUR_STATE, IB_WQ_FLAGS
+modify_wq_flags = IB_WQ_FLAGS_CVLAN_STRIPPING, IB_WQ_FLAGS_SCATTER_FCS, IB_WQ_FLAGS_DELAY_DROP, IB_WQ_FLAGS_PCI_WRITE_END_PADDING
+srq_type = IB_SRQT_BASIC, IB_SRQT_XRC, IB_SRQT_TM
+create_srq_ex_flags = MLX5_SRQ_FLAG_SIGNATURE
+
+# structs
+# =======
+
+# context
+# ===========
+
+mlx5_get_context_cmd_resp {
```

Comment:
The first 2 fields seem to correspond to ib_uverbs_get_context_resp struct. It would be useful to have ib_uverbs_get_context_resp struct explicitly defined and used here. Something along the lines of:

```
type ib_uverbs_get_context_resp[DRIVER_DATA] {
  async_fd			int32
  num_comp_vectors		int32
  driver_data DRIVER_DATA
}

mlx5_get_context_cmd_resp {
  qp_tab_size			int32
  ...
}
```

And then use `ib_uverbs_get_context_resp[mlx5_get_context_cmd_resp]` instantiation for mlx5.
Is there a kernel name for mlx5_get_context_cmd_resp? I can't find it in kernel:
https://elixir.bootlin.com/linux/v5.1-rc1/ident/mlx5_get_context_cmd_resp
If so please use it.